### PR TITLE
Fix verifiable manifest initialization

### PR DESCRIPTION
**Pull request by gmega**
This PR ensures that the interleaving strategy for protected manifests gets passed onto its derived verifiable manifest. This is one of the reasons why we were seeing decoding failures with verifiable manifests in #821.

### CODE REVIEW

File: tests/codex/testmanifest.nim [hunk 1]
```diff
@@ -74,3 +74,28 @@ checksuite "Manifest":
...
+      strategy = SteppedStrategy
+    )
+    verifiable = Manifest.new(
+      manifest = protected,
+      verifyRoot = Cid.example,
+      slotRoots = @[Cid.example, Cid.example]
+    ).tryGet()
+
+  test "Should preserve interleaving strategy for protected manifest in verifiable manifest":
+    check verifiable.protectedStrategy == SteppedStrategy
```
Comment (emizzle) [inline tests/codex/testmanifest.nim:101]: Souls were also check the default is LinearStrategy in the case that the protected manifest has not set a strategy?

**Review by emizzle — APPROVED** [review]

**Review by benbierens — APPROVED** [review]

**Review by tbekas — APPROVED** [review]

File: tests/codex/testmanifest.nim (hunk 1 above)
Comment (gmega) [reply tests/codex/testmanifest.nim:101]: Good point

File: codex/node.nim [hunk 2]
```diff
@@ -559,8 +561,8 @@ proc onStore(
 
     return success()
 
-  without indexer =? manifest.protectedStrategy.init(
-    0, manifest.numSlotBlocks() - 1, manifest.numSlots).catch, err:
+  without indexer =? manifest.verifiableStrategy.init(
```
Comment (dryajov) [inline codex/node.nim:564]: This only needs to download the slot's blocks, not all the blocks.

Comment (gmega) [reply codex/node.nim:564]: It's not downloading all the blocks, it's slicing the required blocks from the space of all blocks.

**Review by dryajov — APPROVED** [review]

**Pull request merged**